### PR TITLE
Retire legacy markup for global header

### PR DESCRIPTION
**Pull request by yozzo**
The global header is inteded to be used across all applications, so the
existing mark-up in data-hub needs to be replaced with the new, resuable
one, that is using the latest GDS design system.

* retire old markup
* remove unused stylesheets related to the previous header

### CODE REVIEW

File: assets/stylesheets/objects/base/_global-header.scss [hunk 1]
```diff
@@ -1,67 +1,163 @@
...
-  &:focus {
-    color: inherit;
+  .global-nav__link {
+    color: govuk-colour("black");
+    background: inherit;
+    font-weight: 700;
+    display: block;
+    text-decoration: none;
+    padding: 7px 0;
+    font-size: 16px;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use the mixin ` @include govuk-font($size: 16)` for this?

File: assets/stylesheets/objects/base/_global-header.scss [hunk 2]
```diff
@@ -1,67 +1,163 @@
...
-}
+  .govuk-header {
+    padding-top: 2px;
+    border-bottom: none;
+  }
+
+  .global-nav {
+    background: govuk-colour("grey-3");
+    margin: 0;
+    padding: 4px 0;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit`

File: assets/stylesheets/objects/base/_global-header.scss [hunk 3]
```diff
@@ -1,67 +1,163 @@
-@import "../../settings/colours";
-@import "../../tools/mixins";
+@import "~govuk-frontend/helpers/all";
+@import "~govuk-frontend/settings/all";
+
+.govuk-header__container {
+  border-bottom: none;
+  margin-bottom: 0;
+  padding-top: 8px;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit` for padding and margin

File: assets/stylesheets/objects/base/_global-header.scss [hunk 4]
```diff
@@ -1,67 +1,163 @@
...
   }
-}
 
-.global-header__inner {
-  @include clearfix;
-  @include site-width-container;
-  padding: 10px 0;
-}
+  .govuk-header {
+    padding-top: 2px;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit` for padding and margin

File: assets/stylesheets/objects/base/_global-header.scss [hunk 5]
```diff
@@ -1,67 +1,163 @@
...
-  &:visited,
-  &:focus {
-    color: inherit;
+  .global-nav__link {
+    color: govuk-colour("black");
+    background: inherit;
+    font-weight: 700;
+    display: block;
+    text-decoration: none;
+    padding: 7px 0;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit` for padding and margin

File: assets/stylesheets/objects/base/_global-header.scss [hunk 6]
```diff
@@ -1,67 +1,163 @@
...
+    font-size: 16px;
+
+    &.active {
+      color: govuk-colour("blue");
+    }
+
+    @include govuk-media-query($from: tablet) {
+      padding: 8px 0;
+      margin-right: 20px;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit` for padding and margin

File: assets/stylesheets/objects/base/_global-header.scss [hunk 7]
```diff
@@ -1,67 +1,163 @@
...
+      font-size: 16px;
+      line-height: 23px;
+
+      &:hover {
+        color: govuk-colour("black");
+        padding: 8px 0 3px 0;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit` for padding and margin

File: assets/stylesheets/objects/base/_global-header.scss [hunk 8]
```diff
@@ -1,67 +1,163 @@
...
+        border-bottom: 5px solid $govuk-link-hover-colour;
+      }
+
+      &.active {
+        padding: 8px 0 3px 0;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit` for padding and margin

File: assets/stylesheets/objects/base/_global-header.scss [hunk 9]
```diff
@@ -1,67 +1,163 @@
...
+  .proposition-menu {
+    float: none;
+    padding-bottom: 9px;
 
-.global-header__site-name {
-  @include visually-hidden;
-}
+    @include govuk-media-query($from: tablet) {
+      float: right;
+      margin: 5px 0;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit` or `$govuk-spacing` for padding and margin

File: assets/stylesheets/objects/base/_global-header.scss [hunk 10]
```diff
@@ -1,67 +1,163 @@
...
+      border-bottom: none;
+    }
   }
-}
+
+  .proposition-menu__link {
+    width: 100%;
+    font-size: 14px;
+    display: block;
+    padding: 10px 0;
```
Comment (ian-leggett) [inline assets/stylesheets/objects/base/_global-header.scss]: Should we use `$default-spacing-unit` or `$govuk-spacing` for padding and margin

File: assets/stylesheets/objects/base/_global-header.scss (hunk 2 above)
Comment (yozzo) [reply assets/stylesheets/objects/base/_global-header.scss]: as discussed, will have to go with pixels in this component. `$govuk-spacing` doesn't give use the values we need, and `$default-spacing-unit` will be retired.

File: assets/stylesheets/objects/base/_global-header.scss (hunk 1 above)
Comment (yozzo) [reply assets/stylesheets/objects/base/_global-header.scss]: as demoed, this mixin changes the font-weights making everything thinner
screenshot 2019-02-08 at 11 55 49

File: assets/stylesheets/objects/base/_global-header.scss (hunk 3 above)
Comment (yozzo) [reply assets/stylesheets/objects/base/_global-header.scss]: as agreed, we'll need to go with pixels for this component, as it uses the legacy design, but the new toolkit,

File: assets/stylesheets/objects/base/_global-header.scss (hunk 1 above)
Comment (yeahfro) [reply assets/stylesheets/objects/base/_global-header.scss]: @yozzo Have you tried passing the `$weight` parameter?

**Pull request merged**